packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPFrameUpdater.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if TARGET_OS_OSX
+#import <FlutterMacOS/FlutterMacOS.h>
+#else
+#import <Flutter/Flutter.h>
+#endif
+
+#import <AVFoundation/AVFoundation.h>
+
+/// FVPFrameUpdater is responsible for notifying the Flutter texture registry
+/// when a new video frame is available.
+@interface FVPFrameUpdater : NSObject
+/// The texture ID associated with the video output.
+@property(nonatomic) int64_t textureId;
+/// The Flutter texture registry used to notify about new frames.
+@property(nonatomic, weak, readonly) NSObject<FlutterTextureRegistry> *registry;
+/// The output that this updater is managing.
+@property(nonatomic, weak) AVPlayerItemVideoOutput *videoOutput;
+/// The last time that has been validated as avaliable according to hasNewPixelBufferForItemTime:.
+@property(nonatomic, assign) CMTime lastKnownAvailableTime;
```

Comment:
Do all of these need to be public?

Reply:
`registry` could be private. `videoOutput` and `lastKnownAvailableTime` are referenced in `FVPVideoPlayer` implementation

Reply:
Yes, clients should definitely not be setting this value.

Reply:
Made it `readonly`